src/realm/sync/noinst/integer_codec.hpp

```diff
@@ -101,6 +105,32 @@ std::size_t encode_int(char* buffer, T value)
     REALM_DIAG_POP();
 }
 
+template <>
+inline std::size_t encode_int(char* buffer, Bid128 value)
+{
+    auto value_0 = value.w[0];
+    auto value_1 = value.w[1];
+    constexpr int bits_per_byte = 7;
+    using uchar = unsigned char;
+    char* ptr = buffer;
+    while (value_0 >> (bits_per_byte - 1)) {
```

Comment:
Seems wrong. 1) why the "bits_per_byte-1", 2) isn't there a risk for too early out since value_1 is not part of the condition.

Reply:
1) "bits_per_byte-1" because there should be room for the sign bit in the last byte (always 0 in this case).
2) An error - well spotted.

Reply:
perhaps add a comment :-)

Reply:
Has tests been updated to catch the error Finn spotted?

Reply:
The error is fixed

Reply:
@jedelbo Yes, but I was more wondering if you added tests that would have found this error instead of Finn :-)